rr_common/src/planner/planner.cpp

```diff
@@ -105,6 +106,18 @@ float aggregateCost(control_vector &controlVector, pcl::KdTreeFLANN<pcl::PointXY
     return costSum;
 }
 
+/**
+ * Member returning the current bestAngle averaged against the median.
+ * Median is calculated based on previous angles, or PREV_STEERING_ANGLES
+*/
+float steeringAngleAverageAgainstMedian(float bestAngle) {
+    PREV_STEERING_ANGLES_INDEX = (PREV_STEERING_ANGLES_INDEX + 1) % SMOOTHING_ARRAY_SIZE;
+    std::vector<float> sortedSteeringAngles(PREV_STEERING_ANGLES);
+    sort(sortedSteeringAngles.begin(), sortedSteeringAngles.end()); //sort array of previous vectors in ascending order to find median
```

Comment:
Is this the only code that uses the sorted nature of sortedSteeringAngles? If so, consider using [`nth_element`](http://en.cppreference.com/w/cpp/algorithm/nth_element). It does a partial sort on the container: just enough to give you a correct answer for what value would be at position N if the container were fully sorted. 

Note that `nth_element` modifies the container it's given, so you'll still have to copy the values into a temporary container as you've done here.